### ISSUE TITLE
Lots of crashes on iOS 8 and iOS 9 when streaming from an HTTPS soundcloud URL

### ISSUE THREAD

**Issue by corymsmith** [muhku/FreeStreamer#236]
Any idea why this might be happening? I can't seem to replicate this on my phone or in the simulator but I'm seeing lots of crashes in Crashlytics (none on iOS7 though). I'm wondering if #216 may help to address this?

```
Crashed: com.apple.NSURLConnectionLoader
EXC_BAD_ACCESS KERN_INVALID_ADDRESS 0x0000000000000000

0 libsystem_coretls.dylib SSLEncodeClientHello + 999
1 libsystem_coretls.dylib SSLEncodeClientHello + 990
2 libsystem_coretls.dylib SSLPrepareAndQueueMessage + 28
3	libsystem_coretls.dylib SSLAdvanceHandshake + 150
4 Security SSLHandshakeProceed + 28
5	Security SSLHandshake + 138
6 CFNetwork SocketStream::_PerformSecurityHandshake_NoLock() + 958
7 CFNetwork SocketStream::socketCallback(__CFSocket*, unsigned long, __CFData const*, void const*) + 128
8 CFNetwork SocketStream::_SocketCallBack_stream(__CFSocket*, unsigned long, __CFData const*, void const*, void*) + 58
9 CoreFoundation __CFSocketPerformV0 + 756
10 CoreFoundation __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 14
11 CoreFoundation __CFRunLoopDoSources0 + 222
12 CoreFoundation __CFRunLoopRun + 768
13 CoreFoundation CFRunLoopRunSpecific + 476
14 CoreFoundation CFRunLoopRunInMode + 106
15 CFNetwork +[NSURLConnection(Loader) _resourceLoadLoop:] + 486
16 Foundation __NSThread__main__ + 1118
17 libsystem_pthread.dylib _pthread_body + 138
18 libsystem_pthread.dylib _pthread_start + 118
19	libsystem_pthread.dylib thread_start + 8

**Comment by muhku** [comment]
Stability fixes done in Issues #238 and #243 should be in. Also Pull Requests #240 and #242 merged. Let's reopen if the problems still reoccur with these in v3.1.1.

**Comment by corymsmith** [comment]
I released a new version and I don't see any of these crashes anymore. I still see bad access errors related to this lib, I'll create some issues when I get a chance to investigate.
